### PR TITLE
[BugFix] refactor cloud native configuration check, fix hdfs support

### LINKED ISSUE #20044: [StarOS] fe cannot started when cloud_native_storage_type = HDFS
### Steps to reproduce the behavior (Required)
9c29dc3d-88e1-4d99-8c7d-0565955e5003

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
BRANCH-3.0-3209C7C
```

### PR DESCRIPTION
**Pull request by xiangguangyxg**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #20044

## Problem Summary(Required) ：

Refactor cloud native configuration check, fix hdfs support.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/staros/StarMgrServer.java [hunk 1]
```diff
@@ -123,37 +123,40 @@ public void initialize(BDBEnvironment environment, String baseImageDir) throws I
 
         // Storage fs type
         com.staros.util.Config.DEFAULT_FS_TYPE = Config.cloud_native_storage_type;
-        if (!com.staros.util.Config.DEFAULT_FS_TYPE.equals("HDFS") && !com.staros.util.Config.DEFAULT_FS_TYPE.equals("S3")) {
+        if (com.staros.util.Config.DEFAULT_FS_TYPE.equals("HDFS")) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/staros/StarMgrServer.java:126]: Better use `equalsIgnoreCase`?

**Review by kevincai — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/staros/StarMgrServer.java [hunk 2]
```diff
@@ -123,37 +123,40 @@ public void initialize(BDBEnvironment environment, String baseImageDir) throws I
...
+            // HDFS related configuration
+            com.staros.util.Config.HDFS_URL = Config.cloud_native_hdfs_url;
+            if (com.staros.util.Config.HDFS_URL.isEmpty()) {
+                LOG.error("HDFS url is empty.");
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/staros/StarMgrServer.java:130]:
```suggestion
                LOG.error("The configuration item \"cloud_native_hdfs_url\" is empty.");
```

**Review by imay — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/staros/StarMgrServer.java (hunk 1 above)
Comment (xiangguangyxg) [reply fe/fe-core/src/main/java/com/starrocks/staros/StarMgrServer.java]: done

File: fe/fe-core/src/main/java/com/starrocks/staros/StarMgrServer.java (hunk 2 above)
Comment (xiangguangyxg) [reply fe/fe-core/src/main/java/com/starrocks/staros/StarMgrServer.java]: done

**Review by sduzh — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**